[Access for SaaS] Braintree SAML tutorial (#15575)
cloudflare/cloudflare-docs: content/cloudflare-one/applications/configure-apps/saas-apps/braintree-saas.md

---
pcx_content_type: how-to
title: Braintree
weight: 5
---

# Connect to Braintree through Access

This guide covers how to configure [Braintree](https://developer.paypal.com/braintree/articles/guides/single-sign-on-sso) as a SAML application in Cloudflare Zero Trust.

## Prerequisites

- A [SAML identity provider](/cloudflare-one/identity/idp-integration/generic-saml/) configured in Cloudflare Zero Trust
- Admin access to a Braintree production or sandbox account

## 1. Add a SaaS application to Cloudflare Zero Trust

1. In [Zero Trust](https://one.dash.cloudflare.com), go to **Access** > **Applications**.
2. Select **Add an application** > **SaaS** > **Select**.

> 3. For **Application**, type `Braintree` and select the textbox that appears below.

Reviewer: ```suggestion
3. For **Application**, enter `Braintree` and select the textbox that appears below.
```

4. For the authentication protocol, select **SAML**.
5. Select **Add application**.
6. Fill in the following fields with temporary values:
    - **Entity ID**: `placeholder`
    - **Assertion Consumer Service URL**: `https://www.placeholder.com`
    - **Name ID format**: _Email_

> 7. Copy the **SSO endpoint** and **Public key**.

Reviewer: Do the other tutorials bold the items you need to copy?

```suggestion
7. Copy the SSO endpoint and Public key.
```

Author: Yes, the other tutorials bold them. In the UI, each item has a bold title with a text field below it containing the content to copy. Since the style guide says bold can be used to highlight a title/name, I've been using it as such.

    - **Email domain(s) to be used in user IDs**: The email domain(s) that should be allowed to sign in to your account via SSO.
    - **Single Sign-on HTTP POST Binding URL**: SSO endpoint from application configuration in Cloudflare Zero Trust
    - **Certificate for validation**: Public key from application configuration in Cloudflare Zero Trust.
4. Select whether you are using a **Production** or **Sandbox** account.
5. Fill out the **Your contact information** fields and select **Submit a help request**.
6. When you receive an email stating SSO has been successfully configured for your account, you can proceed to the next step.

## 3. Finish adding a SaaS application to Cloudflare Zero Trust

1. In [Zero Trust](https://one.dash.cloudflare.com), go to **Access** > **Applications**.
2. Select **Braintree** > **Edit** > **Overview**.
3. Replace the temporary values for **Entity ID** and **Assertion Consumer Service URL** with the link provided in the successful SSO configuration email from Braintree support. You will use the same link for both values.
4. Select **Save Application**.

## 4. Test the integration and add SSO users

> 1. In your Braintree Control Panel, go to the **Settings** icon > **Team**.

Reviewer: ```suggestion
1. In your Braintree Control Panel, select the gear icon > **Team**.
```
per my comment, but waiting for confirmation!
